pkg/kube/pod_test.go

```diff
@@ -62,17 +62,76 @@ func (s *PodSuite) TearDownSuite(c *C) {
 }
 
 func (s *PodSuite) TestPod(c *C) {
+	// get controllers's namespace
+	cns, err := GetControllerNamespace()
+	c.Assert(err, IsNil)
+
+	// get controller's SA
+	sa, err := GetControllerServiceAccount(s.cli)
+	c.Assert(err, IsNil)
+
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 	defer cancel()
-	pod, err := CreatePod(context.Background(), s.cli, &PodOptions{
-		Namespace:    s.namespace,
-		GenerateName: "test-",
-		Image:        "kanisterio/kanister-tools:0.28.0",
-		Command:      []string{"sh", "-c", "tail -f /dev/null"},
-	})
+
+	podOptions := []*PodOptions{
+		{
+			Namespace:    s.namespace,
+			GenerateName: "test-",
+			Image:        "kanisterio/kanister-tools:0.28.0",
+			Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		},
+		{
+			Namespace:          s.namespace,
+			GenerateName:       "test-",
+			Image:              "kanisterio/kanister-tools:0.28.0",
+			Command:            []string{"sh", "-c", "tail -f /dev/null"},
+			ServiceAccountName: "dummy-sa",
+		},
+		{
+			Namespace:    cns,
+			GenerateName: "test-",
+			Image:        "kanisterio/kanister-tools:0.28.0",
+			Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		},
```

Comment:
Should add one more testcase, where - `Namespace: cns` and `ServiceAccountName: "non-default"`.